source/posix/system_info.c

```diff
@@ -445,3 +450,40 @@ enum aws_platform_os aws_get_platform_build_os(void) {
     return AWS_PLATFORM_OS_UNIX;
 }
 #endif /* AWS_OS_APPLE */
+
+#if defined(__linux__) || defined(__unix__)
+static const uint64_t S_BYTES_TO_KILO_BYTES = 1024;
+#endif
+
+int aws_get_memory_usage(int64_t *output) {
+// Get the Memory usage from Linux
+#if defined(__linux__) || defined(__unix__)
+    struct sysinfo memory_info;
+    sysinfo(&memory_info);
+    uint64_t physical_memory_used = memory_info.totalram - memory_info.freeram;
+    physical_memory_used *= memory_info.mem_unit;
+    // Return data in Kilobytes
+    physical_memory_used = physical_memory_used / S_BYTES_TO_KILO_BYTES;
+    *output = physical_memory_used;
+    return AWS_OP_SUCCESS;
+#endif
```

Comment:
trivial: `#else` instead of just #endif here. I could see some compiler complaining that the rest of the code is unreachable